clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4994,6 +4994,10 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-Wno-sycl-strict");
     }
 
+    // Set O2 optimization level by default
+    if (!Args.getLastArg(options::OPT_O_Group))
+      CmdArgs.push_back("-O2");
```

Comment:
This should be OK - could you add a test to verify this is working for both clang and clang-cl type compilations?

Reply:
Good point. I'll add a test.